examples/range_test.go

```diff
@@ -0,0 +1,220 @@
+	err = query.SetLayout(tiledb.TILEDB_ROW_MAJOR)
+	checkError(err)
+	_, err = query.SetBuffer("a", data)
+	checkError(err)
+
+	// Perform the write and close the array.
+	err = query.Submit()
+	checkError(err)
+	err = array.Close()
+	checkError(err)
+}
+
+func addRange() {
+	ctx, err := tiledb.NewContext(nil)
+	checkError(err)
+
+	// Prepare the array for reading
+	array, err := tiledb.NewArray(ctx, rangeArrayName)
+	checkError(err)
+	err = array.Open(tiledb.TILEDB_READ)
+	checkError(err)
+
+	// Prepare the query
+	query, err := tiledb.NewQuery(ctx, array)
+	checkError(err)
+
+	var startFloat float32 = 1
+	var endFloat float32 = 3
+
+	// Try with invalid dimension types
+	err = query.AddRange(0, &startFloat, &endFloat)
+	fmt.Println(err)
+
+	var start int32 = 1
+	var end int32 = 3
+
+	// Try with invalid dimension index
+	err = query.AddRange(2, &start, &end)
+	fmt.Println(err)
+
+	// Try using valid index, range
+	err = query.AddRange(0, &start, &end)
+	checkError(err)
+
+	err = array.Close()
+	checkError(err)
+}
+
+func getRangeNum() {
+	ctx, err := tiledb.NewContext(nil)
+	checkError(err)
+
+	// Prepare the array for reading
+	array, err := tiledb.NewArray(ctx, rangeArrayName)
+	checkError(err)
+	err = array.Open(tiledb.TILEDB_READ)
+	checkError(err)
+
+	// Prepare the query
+	query, err := tiledb.NewQuery(ctx, array)
+	checkError(err)
+
+	// Try using valid index
+	rangeNum, err := query.GetRangeNum(0)
+	checkError(err)
+
+	fmt.Printf("Number of ranges across dimension 0 is: %d\n", *rangeNum)
+
+	err = array.Close()
+	checkError(err)
+}
+
+func getRange() {
+	ctx, err := tiledb.NewContext(nil)
+	checkError(err)
+
+	// Prepare the array for reading
+	array, err := tiledb.NewArray(ctx, rangeArrayName)
+	checkError(err)
+	err = array.Open(tiledb.TILEDB_READ)
+	checkError(err)
+
+	// Prepare the query
+	query, err := tiledb.NewQuery(ctx, array)
+	checkError(err)
+
+	// Try using valid dimension index and range index
+	start, end, err := query.GetRange(0, 0)
+	checkError(err)
+
+	fmt.Printf("Range start for dimension 0, range 0 is: %d\n", start.(int32))
+	fmt.Printf("Range end for dimension 0, range 0 is: %d\n", end.(int32))
+
+	err = array.Close()
+	checkError(err)
+}
+
+// ExampleRange shows an example of creation, writing of a dense array
+// and useage of range functions
+func ExampleRange() {
+	createRangeArray()
+	writeRangeArray()
+	addRange()
+	getRangeNum()
+	getRange()
+
+	// Cleanup example so unit tests are clean
+	if _, err := os.Stat(rangeArrayName); err == nil {
+		err = os.RemoveAll(rangeArrayName)
+		checkError(err)
+	}
+
+	// Output: Error adding query range: [TileDB::Subarray] Error: Cannot add range to dimension; Range must be in the domain the subarray is constructed from
+	// Error adding query range: [TileDB::Subarray] Error: Cannot add range to dimension; Invalid dimension index
+	// Number of ranges across dimension 0 is: 1
+	// Range start for dimension 0, range 0 is: 1
+	// Range end for dimension 0, range 0 is: 4
```

Comment:
I expected this to be 3

Reply:
Ranges are inclusive, and you set `var end int32 = 3`, so I believe 3 is correct.